test_schedule.py

```diff
@@ -17,6 +19,10 @@
     IntervalError,
 )
 
+# Set timezone to Europe/Berlin (CEST) to ensure global reproducibility
+os.environ["TZ"] = "CET-1CEST,M3.5.0,M10.5.0"
+time.tzset()
```

Comment:
Why is this line here? Can we delete it?

Reply:
This line ensures changing the timezone is updated throughout python lib as described [here](https://docs.python.org/3/library/time.html#time.tzset): 
```
Although in many cases, changing the TZ environment variable may affect the output of functions like [localtime()](https://docs.python.org/3/library/time.html#time.localtime) without calling [tzset()](https://docs.python.org/3/library/time.html#time.tzset), this behavior should not be relied on.
```